src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1702,7 +1702,8 @@ private String redirectRelativeLinks(Element element, TextTree tt) {
         if (lower.startsWith("mailto:")
                 || lower.startsWith("http:")
                 || lower.startsWith("https:")
-                || lower.startsWith("file:")) {
+                || lower.startsWith("file:")
+                || lower.startsWith("ftp:")) {
```

Comment:
Adding `ftp:` is OK, but given that the method is about modifying relative URLs, a reasonable/preferable alternative would be to use `URI.isAbsolute`

Reply:
The way to use java.net.URI API has already been discussed above. https://github.com/openjdk/jdk/pull/5198#discussion_r692852885

The method using regex has a security problem because it matches patterns other than ftp. So I adopted @hns 's suggestion.

Reply:
@jonathan-gibbons Could you reply to the above comment?

Reply:
@masyano I think it's ok this way. After all, this is a simple bug and we have spent more than enough time on bike-shedding. If we want to improve the mechanism by using `java.net.URI` we can file a separate issue for it.